ecosystem/sep-0030.md

```diff
@@ -0,0 +1,479 @@
+- The server responds to the client with the signing public key.
+- The client adds the signing public key as a signer of the account with an appropriate weight.
+
+A client can register with multiple servers and give each signing key a weight that is appropriate for the use case the client and servers are targeting. As an example, if a client is intending to use two servers that should individually have no control of the account but together are able to approve transactions, then the client can configure [account thresholds] to high 20, med 20, low 20, and assign weights 10 to both server signing keys.
+
+The signing flow is as follows:
+- The client uses the servers phone or email authentication provider to get a [JWT] proving possession of the phone number or email address.
+- Alternatively, the client uses [SEP-10] to get a [JWT] proving high threshold control of another account that is listed as an alternative identity of the account to be accessed.
+- The client, authenticated as the phone, email, or another account, requests the server to sign a transaction.
+- The server verifies that the [JWT] is valid according to the authentication providers specification.
+- The server verifies that the phone, email, or account in the [JWT] is an alternative identity of the account.
+- The server verifies that the source account of the transaction is the account.
+- The server verifies that the source account of all operations on the transaction are not set or set to the account.
+- The server signs the transaction with the signing secret key.
+- The server responds to the client with the signature.
+
+The client repeats the signing flow with each server it needs to meet the
+accounts threshold to be able to successfully submit a transaction.
+
+The client can use this process to sign a transaction with the [Set Options
+operation] to add a new signer to the account to regain control of an account
+of which they have lost the key, or to gain control of an account shared with
+them.
+
+A client can also use the endpoints in the specification to:
+- Discover registered accounts that it has access to.
+- Update the identities associated with an account.
+- Delete accounts.
+
+## Specification
+
+### Authentication
+
+All endpoints require authentication in the form of a [JWT] token that is passed
+in the `Authorization` header. The [JWT] can be issued by two types of issuers.
+Some endpoints require a specific provider while others should function with
+both. The two issuers are:
+
+- `SEP-10`: A [SEP-10] Web Auth server for proving high threshold control of an
+account. The [SEP-10] server should be configured to only issue [JWT]s if signers
+prove they meet the high threshold on the account.
+- `External`: Defined by the implementer. Proves control of a phone number,
+email address, or other identity.
+
+### Content Type
+
+All endpoints accept in requests the following `Content-Type`s:
+- `application/x-www-form-urlencoded`
```

Comment:
We appear to offer the two content-types in other SEPs like SEP-10. @nebolsin introduced the concept and has shared some points for why it is beneficial, see stellar/stellar-protocol#150.

It's relatively easy for our reference implementations to offer both because [`"github.com/stellar/go/support/http/httpdecode".Decode`](https://godoc.org/github.com/stellar/go/support/http/httpdecode#Decode) functionality that unpacks both in a standard form. I'm not sure if it would be as easy in every programming language though and that's a tradeoff that might not be worth it.

We should maybe try and learn from SEP-10 and find out how many wallets and servers actually support url-encoded form bodies and leave it out if no one is using it.

I've created an issue to consider this: stellar/stellar-protocol#557. If we decide in that issue not to use it going forward I will remove it from this SEP if this SEP is still in draft state.